### PR TITLE
updating fixes to props

### PR DESCRIPTION
**Pull request by iruzevic**
the same as in frontend libs

### CODE REVIEW

**Review by IvanGrginovInf — APPROVED** [review]
Looks good but haven't tested yet.

File: src/Helpers/Components.php [hunk 1]
```diff
@@ -221,41 +221,44 @@ public static function responsiveSelectors(array $items, string $selector, strin
...
-					break;
-				case 'array':
-				case 'object':
-					$defaultValue = isset($manifestKey['default']) ? $manifestKey['default'] : [];
-					break;
-				default:
-					$defaultValue = isset($manifestKey['default']) ? $manifestKey['default'] : '';
-					break;
+		if ($manifestKey === null) {
+			if (array_key_exists('blockName', $manifest)) {
```
Comment (dingo-d) [inline src/Helpers/Components.php:239]: Same as above

File: src/Helpers/Components.php [hunk 2]
```diff
@@ -221,41 +221,44 @@ public static function responsiveSelectors(array $items, string $selector, strin
...
+				throw new \Exception("{$key} key does not exist in the {$manifest['componentName']} component manifest. Please check your implementation. Check if your {$key} attribut exists in the component's manifest.json");
 			}
+		}
+
+		$defaultType = $manifestKey['type'];
 
-			return $defaultValue;
+		switch ($defaultType) {
+			case 'boolean':
+				$defaultValue = isset($manifestKey['default']) ? $manifestKey['default'] : false;
```
Comment (dingo-d) [inline src/Helpers/Components.php:250]:
```suggestion
				$defaultValue = $manifestKey['default'] ?? false;
```

File: src/Helpers/Components.php [hunk 3]
```diff
@@ -221,41 +221,44 @@ public static function responsiveSelectors(array $items, string $selector, strin
...
+				break;
+			case 'array':
+			case 'object':
+				$defaultValue = isset($manifestKey['default']) ? $manifestKey['default'] : [];
```
Comment (dingo-d) [inline src/Helpers/Components.php:254]:
```suggestion
				$defaultValue = $manifestKey['default'] ?? [];
```

File: src/Helpers/Components.php [hunk 4]
```diff
@@ -264,27 +267,30 @@ public static function checkAttr(string $key, array $attributes, array $manifest
...
 	{
 		$output = [];
 
 		if (!isset($manifest['responsiveAttributes'])) {
-			throw new \Exception("It looks like you are missing the responsiveAttributes key in your {$componentName} manifest.");
+			if (array_key_exists('blockName', $manifest)) {
+				throw new \Exception("It looks like you are missing responsiveAttributes key in your {$manifest['blockName']} block manifest.");
+			} else {
+				throw new \Exception("It looks like you are missing responsiveAttributes key in your {$manifest['componentName']} component manifest.");
+			}
```
Comment (dingo-d) [inline src/Helpers/Components.php:285]: Same as the comment for the unnecessary else.

File: src/Helpers/Components.php [hunk 5]
```diff
@@ -762,7 +768,11 @@ public static function props(array $attributes, string $realName, string $newNam
 		$parent = $attributes['parent'] ?? '';
 
 		// Replace stuff if there is any changing of the attribute names.
-		if ($parent !== $newNameInternal && $realName !== $newNameInternal) {
+		if (
+			$parent !== $newNameInternal &&
+			$realName !== $newNameInternal &&
+			array_key_exists($newNameInternal, $globalData['components'])
```
Comment (dingo-d) [inline src/Helpers/Components.php:774]: isset || array_key_exists comment

**Review by dingo-d — CHANGES_REQUESTED** [review]
Left some comments.

**Comment by iruzevic** [comment]
@dingo-d I have fixed the all you mentioned. I would leave the unnecessary else's for now because we have the same code on the Js side that I would like to have in sync. Once we make tests for JS part then we can make optimisations for the both codes. agree?

**Review by dingo-d — APPROVED** [review]

**Pull request merged**